ListUtil.xs

```diff
@@ -1238,15 +1256,46 @@ CODE:
                  * use a comparable format, so just use the raw bytes, adding
                  * 'f' to ensure not matching a stringified number */
                 else if (nv_arg < (NV)IV_MIN || nv_arg > (NV)UV_MAX) {
-                    sv_setpvn(keysv, (char *) &nv_arg, sizeof(NV));
+                    sv_setpvn(keysv, (char *) &nv_arg, 8);  /* sizeof(NV) == 8 */
                     sv_catpvn(keysv, "f", 1);
                 }
                 /* smaller floats get formatted using %g and could be equal to
                  * a UV or IV */
                 else {
-                    sv_setpvf(keysv, "%0.*" NVgf, NV_MAX_PRECISION, nv_arg);
+#ifdef WIN32_PERL_NO_ANSI
+
+                   /* Because perl was not built with ansi compliance, doing: *
+                    * sv_setpvf(keysv, "%0.20" NVgf, nv_arg)                  *
+                    * will not always work as intended.                       *
+                    * But the following workaround does what we want.         */
+
```

Comment:
As it's the first statement inside the `{` block, the declaration of `buffer` could be made here, inside the `#ifdef`, reducing its scope to where it is used and overall neater

Reply:
Cool .... and done.
All tests still passing.